Mypy does not allow using the root validator with arguments other than pre or _func

# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.4
            pydantic compiled: True
                 install path: /home/[username]/Desktop/[project]/env/lib/python3.6/site-packages/pydantic
               python version: 3.6.9 (default, Nov  7 2019, 10:44:02)  [GCC 8.3.0]
                     platform: Linux-5.3.0-26-generic-x86_64-with-Ubuntu-18.04-bionic
     optional deps. installed: ['typing-extensions']

```
Code:
```py
    @root_validator(pre=False, skip_on_failure=True)
    def test_validator(cls, values):
        return values
```
Output:
```
test.py:5: error: No overload variant of "root_validator" matches argument types "bool", "bool"
test.py:5: note: Possible overload variants:
test.py:5: note:     def root_validator(_func: Callable[..., Any]) -> classmethod
test.py:5: note:     def root_validator(*, pre: bool = ...) -> Callable[[Callable[..., Any]], classmethod]
```

--- a/changes/1192-samuelcolvin.md
+++ b/changes/1192-samuelcolvin.md
@@ -1,0 +1,1 @@
+fix mypy signature for `root_validator`

--- a/pydantic/class_validators.py
+++ b/pydantic/class_validators.py
@@ -103,7 +103,9 @@ def root_validator(_func: AnyCallable) -> classmethod:
 
 
 @overload
-def root_validator(*, pre: bool = False) -> Callable[[AnyCallable], classmethod]:
+def root_validator(
+    *, pre: bool = False, allow_reuse: bool = False, skip_on_failure: bool = False
+) -> Callable[[AnyCallable], classmethod]:
     ...
 
 

--- a/tests/mypy/modules/success.py
+++ b/tests/mypy/modules/success.py
@@ -33,7 +33,7 @@ class Model(BaseModel):
     def root_check(cls, values: Dict[str, Any]) -> Dict[str, Any]:
         return values
 
-    @root_validator(pre=True)
+    @root_validator(pre=True, allow_reuse=False, skip_on_failure=False)
     def pre_root_check(cls, values: Dict[str, Any]) -> Dict[str, Any]:
         return values
 

